src/mergecil.ml

```diff
@@ -1341,7 +1341,7 @@ let oneFilePass2 (f: file) =
         vi (* Already done *)
       else begin
         (* Maybe it is static. Rename it then *)
-        if vi.vstorage = Static then begin
+        if vi.vstorage = Static || (not !merge_inlines && hasAttribute "gnu_inline" (typeAttrs vi.vtype)) then begin
```

Comment:
None of that code deals with inlines though `not fdec'.svar.vinline`, and it makes sense to treat these functions as if they were static when the `gnu_inline` attribute is set.

Reply:
The code I linked is in the else branch of the actual inlines merging. And the else branch of this at the very end is where everything else is considered, so it seems to me like the fix should be there.

The bigger issue with it being in `processVarinfo` is related to #84 (where variables get merged, but functions don't), just in the other direction (here we avoid merging variables with `__gnu_inline__`, but the renaming is distinct from the merging itself). The logic of whether to merge or not is duplicated and different between `processVarinfo` and below, where I linked, and this is just deepening the difference. It's not helping anything to avoid inconsistencies.